src/services/codefixes/fixMissingCallInCondition.ts

```diff
@@ -0,0 +1,45 @@
+/* @internal */
+namespace ts.codefix {
+    const fixId = "fixMissingCallInCondition";
+    const errorCodes = [
+        Diagnostics.This_condition_will_always_return_true_since_the_function_is_always_defined_Did_you_mean_to_call_it_instead.code,
+    ];
+
+    registerCodeFix({
+        errorCodes,
+        fixIds: [fixId],
+        getCodeActions(context) {
+            const { sourceFile, span } = context;
+            const callName = getCallName(sourceFile, span.start);
+            if (!callName) return;
+
+            const changes = textChanges.ChangeTracker.with(context, t => doChange(t, context.sourceFile, callName));
+            return [createCodeFixAction(fixId, changes, Diagnostics.Add_missing_call, fixId, Diagnostics.Add_all_missing_calls)];
+        },
+        getAllCodeActions: context => codeFixAll(context, errorCodes, (changes, diag) => {
+            const callName = getCallName(diag.file, diag.start);
+            if (callName) doChange(changes, diag.file, callName);
+        })
+    });
+
+    function doChange(changes: textChanges.ChangeTracker, sourceFile: SourceFile, name: Identifier | PrivateIdentifier): void {
+        changes.replaceNode(sourceFile, name, createCall(createIdentifier(name.text), /*typeArguments*/ undefined, /*argumentsArray*/ undefined));

```

Comment:
I can't think of a case where this won't work (as demonstrated by your lovely test cases below), but this syntax tree is technically incorrect and could easily break in the future.  The actual expression of the new call should be the one identified by `context.span` above, not just the last identifier.  For example, in crude pseudo-code, `a.b()` should be `call(a.b)`, not `a.call(b)`.  That's why you had to (effectively) cast away `PrivateIdentifier` here.

Reply:
`textChanges` technically just does string replacements, so replacing the string at `name` with a call to `name` is "technically" ok (since we're not actually building up a tree). If this were a tree mutation, you would need wrap the `createCall​​​​​​​` around the entire access expression. Either way, you should be able to reuse `name` here.

Reply:
@amcasey @rbuckton Thanks for the feedback :+1:.